.github/workflows/ci-pipeline.yml

```diff
@@ -62,4 +62,26 @@ jobs:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
         if: ${{ always() }}
+  publish:
+    name: Publish to Docker Hub and deploy to Azure
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push' && github.event.schedule != '08 13 * * *'
+    steps:
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+      -
+        name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/todo-app:prod
```

Comment:
I'd definitely like to see us specify the `target` explicitly here - while you get away with is because the `production` stage is the final one defined in the Dockerfile, that feels like a convenient coincidence rather than something I trust will always be the case! E.g. if someone added a new stage for a new form of testing/linting, then this step would start pushing that image!